Detections/SigninLogs/AnomalousIPUsageFollowedByTeamsAction.kql

```diff
@@ -0,0 +1,108 @@
+id: 2b701288-b428-4fb8-805e-e4372c574786
+name: Anomalous login followed by Teams action
+description: |
+  'Detects anomalous IP address usage by user accounts and then checks to see if a suspicious Teams action is performed.
+  Query calcualtes IP usage Delta for each user account and selects accounts where a delta >= 90% is observed between the most and least used IP.
+  To further reduce results the query performs a prevalence check on the lowest used IP's country, only keeping IP's where the country is unusual for the tenant (dynamic ranges)
+  Finally the user accounts activity within Teams logs is checked for suspicious commands (modifying user privilages or admin actions) during the period the suspicious IP was active.'
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - SigninLogs
+  - connectorId: CustomConnector
```

Comment:
One thing here, this should be changed to support the upcoming Teams data in OfficeActivity.  Should be:

requiredDataConnectors:
  \- connectorId: Office365
    dataTypes:
      \- OfficeActivity

Reply:
Updated